src/app/product-add/product-add.component.ts

```diff
@@ -12,38 +15,32 @@ export class ProductAddComponent implements OnInit {
     name : new FormControl('', Validators.required),
     category: new FormControl('', Validators.required),
     image: new FormControl('', Validators.required),
-    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[1-9]*$")]),
+    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[0-9]*$")]),
     description: new FormControl('', Validators.required)
   });
 
-  constructor(private productService: ProductsService) { }
+  constructor(private location: Location, private store: Store<AppState>) { }
 
   ngOnInit(): void {
   }
 
-  refreshForm() {
-    this.addForm.setValue({
-      name: "",
-      category: "",
-      image: "",
-      price: "",
-      description: ""
-    });
+  onSubmit() : void {
+    if ( this.addForm.value.name && this.addForm.value.price && this.addForm.value.category && this.addForm.value.image && this.addForm.value.description) {
+      const newProduct = {
```

Comment:
I'd rather you rewrite this as `const newProduct = { ...this.addForm.value, id: Math.random() };` The spread operator will assign the properties from the spread object into the object you're assigning to.